Unit tests fail when there is a plugin 'foo' installed locally

**Bug description**

Unit tests started failing for me locally.  After some troubleshooting I found that it was the presence of a plugin named `foo` installed on my machine that was the cause.  This means the unit tests were somehow playing around with my local configuration instead of a temporary one.  The test in question is `TestDeletePlugin`

**Expected behavior**

Unit tests should not be affected by my local configuration

**Steps to reproduce the bug / Relevant debug output**

```
# The tests pass when I do NOT have a foo plugin installaed
$ go test -timeout 30s -run ^TestDeletePlugin$ github.com/vmware-tanzu/tanzu-cli/pkg/command
ok  	github.com/vmware-tanzu/tanzu-cli/pkg/command	24.103s

# Create a foo dir, as if a plugin was installed in my LOCAL setup
$ mkdir ~/Library/Application\ Support/tanzu-cli/foo
$ go test -timeout 30s -run ^TestDeletePlugin$ github.com/vmware-tanzu/tanzu-cli/pkg/command
ok  	github.com/vmware-tanzu/tanzu-cli/pkg/command	(cached)

# Add --count=1 to prevent go test caching
# and see the tests fail
$ go test -timeout 30s -run ^TestDeletePlugin$ github.com/vmware-tanzu/tanzu-cli/pkg/command --count=1
[i] The tanzu cli essential plugins have not been installed and are being installed now. The install may take a few seconds.

[i] The tanzu cli essential plugins have not been installed and are being installed now. The install may take a few seconds.

[ok] successfully deleted plugin 'foo'
--- FAIL: TestDeletePlugin (7.61s)
    --- FAIL: TestDeletePlugin/delete_an_installed_plugin (3.74s)
        plugin_test.go:246:
            	Error Trace:	/Users/kmarc/git/tanzu-cli/pkg/command/plugin_test.go:246
            	Error:      	Not equal:
            	            	expected: false
            	            	actual  : true
            	Test:       	TestDeletePlugin/delete_an_installed_plugin
FAIL
FAIL	github.com/vmware-tanzu/tanzu-cli/pkg/command	10.268s
FAIL
```

**Output of `tanzu version`**

Today's main: e4e6b74ea

**Environment where the bug was observed (cloud, OS, etc)**

Local

--- a/pkg/command/plugin_test.go
+++ b/pkg/command/plugin_test.go
@@ -123,7 +123,7 @@ func TestPluginList(t *testing.T) {
 			for i, pluginName := range spec.plugins {
 				err = setupFakePlugin(dir, pluginName, spec.versions[i], plugin.SystemCmdGroup, completionType, spec.targets[i], 1, false, []string{pluginName[:2]})
 				assert.Nil(err)
-				pluginInstallationPath = filepath.Join(common.DefaultPluginRoot, pluginName)
+				pluginInstallationPath = filepath.Join(dir, pluginName)
 				pi := &cli.PluginInfo{
 					Name:             pluginName,
 					Description:      fmt.Sprintf("some %s description", pluginName),
@@ -211,25 +211,28 @@ func TestDeletePlugin(t *testing.T) {
 		var completionType uint8
 		t.Run(spec.test, func(t *testing.T) {
 			assert := assert.New(t)
-			cc, err := catalog.NewContextCatalogUpdater("")
+			cupdater, err := catalog.NewContextCatalogUpdater("")
 			assert.Nil(err)
 
+			var pluginInstallationPath string
 			for i, pluginName := range spec.plugins {
 				err = setupFakePlugin(dir, pluginName, spec.versions[i], plugin.SystemCmdGroup, completionType, spec.targets[i], 1, false, []string{pluginName[:2]})
 				assert.Nil(err)
+
+				pluginInstallationPath = filepath.Join(dir, pluginName)
 				pi := &cli.PluginInfo{
 					Name:             pluginName,
 					Description:      fmt.Sprintf("some %s description", pluginName),
 					Group:            plugin.SystemCmdGroup,
 					Aliases:          []string{pluginName[:2]},
 					Version:          spec.versions[i],
-					InstallationPath: filepath.Join(common.DefaultPluginRoot, pluginName),
+					InstallationPath: pluginInstallationPath,
 				}
 				assert.Nil(err)
-				err = cc.Upsert(pi)
+				err = cupdater.Upsert(pi)
 				assert.Nil(err)
 			}
-			cc.Unlock()
+			cupdater.Unlock()
 			rootCmd, err := NewRootCmd()
 			assert.Nil(err)
 			rootCmd.SetArgs(spec.args)
@@ -239,11 +242,19 @@ func TestDeletePlugin(t *testing.T) {
 			if spec.expectedErrorMsg != "" {
 				assert.Contains(err.Error(), spec.expectedErrorMsg)
 			}
+
+			// Need to get a new catalog because the rootCmd.Execute()
+			// updated the catalog file
+			creader, err := catalog.NewContextCatalog("")
+			assert.Nil(err)
+
 			if !spec.expectedFailure {
-				pi, exists := cc.Get(spec.args[2])
-				assert.Equal(exists, true)
-				_, err := os.Stat(pi.InstallationPath)
-				assert.Equal(os.IsNotExist(err), true)
+				_, exists := creader.Get(spec.args[2])
+				assert.Equal(exists, false)
+
+				// tanzu plugin delete does not remove the binary
+				_, err := os.Stat(pluginInstallationPath)
+				assert.Nil(err)
 			}
 		})
 		os.Unsetenv("TEST_CUSTOM_CATALOG_CACHE_DIR")

